unblob/dependencies.py

```diff
@@ -0,0 +1,57 @@
+from shutil import which
+from typing import Dict, List, Tuple
+
+from click import Context, Option, echo
+
+from .handlers import ALL_HANDLERS_BY_PRIORITY
+
+INSTALLED = "\u2713"
+NOT_INSTALLED = "\u2717"
+
+
+def get_dependencies() -> List[str]:  # pragma: no cover
+    return sorted(
+        set(
+            [
+                h._get_extract_command()
+                for handlers in ALL_HANDLERS_BY_PRIORITY
+                for h in handlers
+            ]
+        )
+    )
+
+
+def _is_installed(command: str) -> bool:
+    return bool(which(command))
+
+
+def _get_dependency_map() -> Dict[str, bool]:
+    return {cmd: _is_installed(cmd) for cmd in get_dependencies()}
+
+
+def _pretty_format_dependencies() -> Tuple[int, str]:
+    return_code = 0
```

Comment:
I would do both of these steps here, because here we decide which `mark` should we print, based on if the dependency installed or not. Since we have this knowledge here, we can return with it and not make the caller call an other function to decide on the exit code the very same way this function does.

Maybe the naming is not right, I will try to rewrite this a little bit, but I can't see a good logic to slip this up. After I push a new version, we check that if it satisfy your requirements or not.